src/lib/controllers/environment-file/environmentEditor.ts

```diff
@@ -0,0 +1,300 @@
+				type: 'input',
+				name: 'bastionUsername',
+				default: get(defaultBastion, 'bastionUsername'),
+				validate: inquirerValidateString,
+			},
+			{
+				message: 'Please enter your bastion key path',
+				type: 'input',
+				name: 'bastionKey',
+				default: get(defaultBastion, 'bastionKey'),
+				validate: inquirerValidatePath,
+			},
+			{
+				message: 'Please enter your bastion key password (if any)',
+				type: 'input',
+				name: 'bastionKeyPassword',
+				default: get(defaultBastion, 'bastionKeyPassword'),
+			},
+		];
+		return inquirer.prompt(questions);
+	}
+
+	verbose: boolean;
+	environment: Environment;
+	configurationPath: string;
+
+	public constructor(args: EnvironmentEditorArgs) {
+		const { configurationPath, verbose = false } = args;
+		this.configurationPath = configurationPath;
+		this.verbose = verbose;
+
+		// A default environment for some default suggestions It might be replaced below.
+		this.environment = new Environment({
+			name: 'my-environment',
+			templates: './deploy-templates/',
+			archiveStore: './archive-store',
+			deployTarget: {
+				kubernetesNamespace: 'default',
+				kubernetesAPI: 'kubernetes',
+			},
+			configStore: {
+				kubernetesNamespace: 'default',
+				kubernetesAPI: 'kubernetes',
+			},
+		});
+
+		try {
+			if (configurationPath) {
+				this.environment = loadFromFileSync(configurationPath) as Environment;
+				if (this.verbose) {
+					console.log('loaded:', configurationPath);
+				}
+			} else if (args.environment) {
+				this.environment = args.environment;
+			}
+		} catch (err) {
+			if (err.code === 'ENOENT') {
+				if (this.verbose) {
+					console.debug(err.message);
+				}
+			} else {
+				throw err;
+			}
+		}
+	}
+
+	async inquire() {
+		const questions = [
+			{
+				message: 'Please enter name of the environment',
+				type: 'input',
+				default: this.environment.name,
+				name: 'name',
+				validate: inquirerValidateString,
+			},
+			{
+				message:
+					'Please enter the deploy-templates URI. (You may also use a relative path)',
+				type: 'input',
+				default: this.environment.templates,
+				name: 'templates',
+				validate: inquirerValidateDirectory, // TODO: support git
+			},
+			{
+				message:
+					'Please enter the archive-store URI. (You may also use a relative path)',
+				type: 'input',
+				default: this.environment.archiveStore,
+				name: 'archiveStore',
+				validate: inquirerValidateDirectory, // TODO: support git
+			},
+		];
+		const answers = await inquirer.prompt(questions);
+		const configStore = await this.inquireConfigStore(
+			get(this.environment, 'configStore'),
+		);
+		const deployTarget = await this.inquireDeployTarget(
+			get(this.environment, 'deployTarget'),
+		);
+		this.environment = new Environment(merge(answers, {
+			configStore,
+			deployTarget,
+		}) as EnvironmentObject);
+		return this.environment;
+	}
+
+	async inquireConfigStore(defaultConfigStore: ConfigStore): Promise<object> {
```

Comment:
Should this also be `Promise<ConfigStore>`?

Reply:
Fixed